singer_sdk/streams/core.py

```diff
@@ -980,10 +998,11 @@ def finalize_state_progress_markers(self, state: dict | None = None) -> None:
             context: dict | None
             for context in self.partitions or [{}]:
                 state = self.get_context_state(context or None)
-                finalize_state_progress_markers(state)
-            return
+                self._finalize_state_progress_markers(state)
+        else:
+            self._finalize_state_progress_markers(state)
 
-        finalize_state_progress_markers(state)
+        self._write_state_message()
```

Comment:
Is it possible to call this outside of this scope. This method should _prepare_ the state to be emitted but it shouldn't actually write the message.

Reply:
Nevermind